package.json

```diff
@@ -42,32 +42,47 @@
       "email": "marco.luethy@gmail.com"
     }
   ],
-  "main": "index.js",
-  "dependencies": {
-    "basic-auth": "^2.0.0",
-    "bluebird": "^3.5.1",
-    "lodash": "^4.17.10",
-    "promisify-any": "^2.0.1",
-    "statuses": "^1.5.0",
-    "type-is": "^1.6.16"
-  },
-  "devDependencies": {
-    "eslint": "^5.2.0",
-    "mocha": "^5.2.0",
-    "should": "^13.2.3",
-    "sinon": "^6.1.4"
-  },
-  "license": "MIT",
+  "main": "index.ts",
   "engines": {
-    "node": ">=6.14"
+    "node": ">=8.10"
   },
+  "license": "MIT",
   "scripts": {
-    "pretest": "eslint lib test --fix",
-    "test": "NODE_ENV=test mocha 'test/**/*_test.js'",
-    "test-debug": "NODE_ENV=test mocha  --inspect --debug-brk 'test/**/*_test.js'"
+    "lint": "tslint -p tsconfig.build.json -c tslint.json",
+    "lint:all": "tslint -p tsconfig.json -c tslint.json",
+    "build:clean": "npx shx rm -rf ./dist",
+    "build": "npx npm-run-all build:clean build:test:*",
+    "build:test:src": "npx tsc -p tsconfig.json",
+    "build:test:package": "node ./scripts/build-prod.js",
+    "build:prod": "npx npm-run-all build:clean build:prod:*",
+    "build:prod:src": "npx tsc -p tsconfig.build.json",
+    "build:prod:package": "node ./scripts/build-prod.js",
+    "test": "npx mocha 'test/**/*.spec.ts' --config 'test/.mocharc.jsonc'"
   },
   "repository": {
-    "type": "git",
-    "url": "https://github.com/oauthjs/node-oauth2-server.git"
+    "url": "https://github.com/oauthjs/node-oauth2-server.git",
+    "type": "git"
+  },
+  "devDependencies": {
+    "@types/basic-auth": "^1.1.2",
```

Comment:
`@types` dependencies should (generally) be in the same block as the dependency they provide types for. 

See https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies:
> Our package exposes declarations from each of those, so any user of our browserify-typescript-extension package needs to have these dependencies as well. For that reason, we used "dependencies" and not "devDependencies", because otherwise our consumers would have needed to manually install those packages. If we had just written a command line application and not expected our package to be used as a library, we might have used devDependencies.